lib/misc/dist_calc.dart

```diff
@@ -9,20 +14,37 @@ class DistCalculator {
   final double _conversion = 1 / 111111;
   final Random rng = Random();
 
-  LatLng calculate(int minDist, int maxDist, LatLng usr) {
+  Marker initiateChallenge(int minDist, int maxDist, LatLng? position) {
```

Comment:
Not sure how this is used, but you're saying that position can be null, then that position will never be null